packages/vision/src/helpers.js

```diff
@@ -0,0 +1,197 @@
+   *   The function which will be called with the result of the API call.
+   *
+   *   The second parameter to the callback is an object representing
+   *   [AnnotateImagesResponse]{@link AnnotateImagesResponse}.
+   * @return {Promise} - The promise which resolves to an array.
+   *   The first element of the array is an object representing
+   *   [AnnotateImagesResponse]{@link AnnotateImagesResponse}.
+   *   The promise has a method named "cancel" which cancels the ongoing
+   *   API call.
+   *
+   * @example
+   * var vision = require('@google-cloud/vision').v1();
+   * var request = {
+   *   image: {source: {image_uri: 'gs://path/to/image.jpg'}},
+   *   features: [],
+   * };
+   * client.annotateImage(request).then(response => {
+   *   // doThingsWith(response);
+   * }).catch(function(err) {
+   *   console.error(err);
+   * });
+   */
+  methods.annotateImage = promisify(function(request, options, callback) {
+    // If a callback was provided and options were skipped, normalize
+    // the argument names.
+    if (is.undefined(callback) && is.function(options)) {
+      callback = options;
+      options = undefined;
+    }
+
+    // If there is no image, throw an exception.
+    if (is.undefined(request.image)) {
+      throw new Error('Attempted to call `annotateImage` with no image.');
+    }
+
+    // If we got a filename for the image, open the file and transform
+    // it to content.
+    return coerceImage(request.image, (err, image) => {
+      if (err) {
+        return callback(err);
+      }
+      request.image = image;
+
+      // Call the GAPIC batch annotation function.
+      return this.batchAnnotateImages([request], options, (err, r) => {
+        // If there is an error, handle it.
+        if (err) {
+          return callback(err);
+        }
+
+        // We are guaranteed to only have one response element, since we
+        // only sent one image.
+        var response = r.responses[0];
+
+        // Fire the callback if applicable.
+        return callback(undefined, response);
+      });
+    });
+  });
+
+  // Get a list of features available on the API, and iterate over them
+  // and create single-feature methods for each.
+  var features = gax.grpc().load([{
+    root: protoFiles('..'),
+    file: `google/cloud/vision/${apiVersion}/image_annotator.proto`,
+  }]).google.cloud.vision[apiVersion].Feature.Type;
+  for (let feature of Object.keys(features)) {
+    // Determine the method name that should be used for the helper method,
+    // based on the Feature.Type enum in the proto.
+    //
+    // The method name should be camelCased (e.g. "FACE_DETECTION" becomes
+    // "faceDetection").
+    let featureValue = features[feature];
+    let methodName = feature.toLowerCase().replace(/_([\w])/, match => {
+      return match[1].toUpperCase();
+    });
+
+    // Assign the single-feature method to the `methods` object.
+    methods[methodName] = promisify(_createSingleFeatureMethod(featureValue));
```

Comment:
Using a loop to define methods is a problem for the docs. I'm not sure if this will work, we might have to actually write out the methods in order to get them into the docs.